FlowCrypt/Functionality/Api/Remote Pub Key Apis/PubLookup.swift

```diff
@@ -69,6 +69,12 @@ class PubLookup: PubLookupType {
 
     func fetchRemoteUpdateLocal(with recipient: Recipient) async throws -> RecipientWithSortedPubKeys {
         let recipient = try await self.lookup(recipient: recipient)
+        // Return local contact(with revoked key) if local stored key is revoked key and fingerprints are same
+        if let contact = try await localContactsProvider.searchRecipient(with: recipient.email),
+           contact.keyState == .revoked,
+           contact.primaryFingerprint == recipient.primaryFingerprint {
```

Comment:
Changed logic as follows. Let me know your thought

Reply:
@tomholub

Reply:
See other comment. I really think this should be removed. If it does have any effect at all (I don't think it does), then it should be achieved differently / in another place.

Reply:
That's why we have revoked key state check here. 

```
contact.keyState == .revoked,
contact.pubKeys == recipient.pubKeys
```

Reply:
So here's logic.

1. We get recipient info from remote.
2. Get locally saved recipient
3. If locally saved recipient key state is revoked & locally saved recipient pubkeys & remote recipient pubkeys are same, then use local saved recipient (with revoke status & revoke key user name) to display recipient in recipient 
input.
4. If not, then use new remote recipient info to display recipient in recipient input

Reply:
> That's why we have revoked key state check here.
> 
> ```
> contact.keyState == .revoked,
> contact.pubKeys == recipient.pubKeys
> ```

Doesn't change what I said. 

swift
```
if a == b && b.revoked == true {
  return b
}
return a
```

can be written as

```swift
return a
```

Same effect, provided that `a == b` means `a deep equals b`

Reply:
> If locally saved recipient key state is revoked & locally saved recipient pubkeys & remote recipient pubkeys are same, then use local saved recipient (with revoke status & revoke key user name) to display recipient in recipient input.

if locally saved pubkeys are the same as received pubkeys, then it doesn't matter whether you render local or remote. They are the same. Regardless whether one of them is revoked or not.

Reply:
I think we are on the different line.
Current logic is not same as this one.
```
if a == b && b.revoked == true {
  return b
}
return a
```

I added comments at every lines to let everyone better understands.
 
```
let recipient = try await self.lookup(recipient: recipient) // This is recipient info from remote
if let contact = try await localContactsProvider.searchRecipient(with: recipient.email), // First check if we have local saved contact
  contact.keyState == .revoked, // Check if local contact key is revoked because if local contact key state is revoked, we should not use further keys with same fingerprints. Instead we should use local revoked key from now on
  contact.pubKeys == recipient.pubKeys { // Even local saved key is revoked key, we should use newer(remote) key if fingerprints(pubkeys) are not same
   return contact
}
try localContactsProvider.updateKeys(for: recipient)
return recipient

```

@tomholub please check

Reply:
I think we are on the different line.
Current logic is not equal to this one.

```
if a == b && b.revoked == true {
  return b
}
return a
```

I added comments to let everyone better understands.

```
let remoteRecipient = try await self.lookup(recipient: recipient) // This is recipient info from remote
if let localContact = try await localContactsProvider.searchRecipient(with: recipient.email), // First check if we have local saved contact
  localContact.keyState == .revoked, // Check if local contact key is revoked because if local contact key state is revoked, we should not use further keys with same fingerprints. Instead we should use local revoked key from now on
  localContact.pubKeys == remoteRecipient.pubKeys { // Even local saved key is revoked key, we should use newer(remote) key if fingerprints(pubkeys) are not same
   return localContact
}
try localContactsProvider.updateKeys(for: remoteRecipient)
return remoteRecipient
```

@tomholub Please check